### PR TITLE
Add `MSG_WAITFORONE` to freebsd and openbsd

### PR DESCRIPTION
**Pull request by stevenengler**
Adds `MSG_WAITFORONE` to freebsd, openbsd, and ~illumos~, as requested in https://github.com/nix-rust/nix/pull/2014.

I got the values from:

- freebsd: http://fxr.watson.org/fxr/source/sys/socket.h#L473
- openbsd: https://github.com/openbsd/src/blob/2852e11abfc574a216ce741308fb0c6968d9617a/sys/sys/socket.h#L512
- illumos: https://github.com/illumos/illumos-gate/blob/717646f7112314de3f464bc0b75f034f009c861e/usr/src/boot/sys/sys/socket.h#L434

This flag is also supposedly supported on [solaris](https://docs.oracle.com/cd/E88353_01/html/E37843/recvmmsg-3c.html), but I don't know how to find the value. If anyone knows what it is, I'll add it to the PR.

### CODE REVIEW

**Review by pfmooney — CHANGES_REQUESTED** [review]
As noted, this constant should not be defined for illumos.

File: src/unix/solarish/illumos.rs [hunk 1]
```diff
@@ -33,6 +33,8 @@ pub const TCP_KEEPCNT: ::c_int = 35;
 pub const TCP_KEEPINTVL: ::c_int = 36;
 pub const TCP_CONGESTION: ::c_int = 37;
 
+pub const MSG_WAITFORONE: ::c_int = 0x80000;
```
[the remark this answers is not in the document]
Comment (stevenengler) [reply src/unix/solarish/illumos.rs:36]: Okay I'll remove it. Thanks!

**Review by semarie — APPROVED** [review]
LGTM for openbsd part. thanks.

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
Thanks for the PR and reviewing! @bors r+

**Comment by stevenengler** [comment]
Not sure I'll have permission for this: @bors retry

**Pull request merged**